app/components/candidate_interface/decoupled_references_review_component.rb

```diff
@@ -29,6 +29,15 @@ def can_send?(reference)
         ).valid?
     end
 
+    def can_resend?(reference)
+      (reference.cancelled? || reference.cancelled_at_end_of_cycle? || reference.email_bounced?) &&
+        !reference.application_form.enough_references_have_been_provided? &&
+        CandidateInterface::Reference::SubmitRefereeForm.new(
+          submit: 'yes',
+          reference_id: reference.id,
+        ).valid?
+    end
```

Comment:
Is it possible to continuously cycle through cancel/resend on a reference, thus spamming the provider? There wouldn't be a sensible reason to do this, but is it worth guarding against it just in case?

Reply:
@malcolmbaig Do you think an `ApplicationReference#resent_at` attribute together with a sensible minimum time limit between resends would solve the problem? Can you think of a simpler solution?

Reply:
Given the low likelihood, let's review if this is a reported problem.